File: CodeVita/prime.c
#include <stdio.h>
#include <stdbool.h>

int gcd(int a, int b){
    while (a != b)
    {
        if(a > b){
            a = a-b;
        }
        else{
            b = b - a;
        }
        return a;
    }
      
}

int lcm(int a, int b){
    return (a*b)/gcd(a,b);
}

bool isPrime(int n){
    if(n == 1){
        return false;
    }
    if(n == 2 || n == 3){
        return true;
    }
    if(n%2 == 0 || n%3 == 0){
        return false;
    }
    for (int i = 5; i*i <= n; i += 6)
    {
        if(n%i == 0 || n%(i+2) == 0){
            return false;
        }
    }
    return true;
}

int main(){
    int arr[11];
    int i = 0;
    do{
        scanf("%d",&arr[i++]);
    }while(getchar() != '\n' && i < 11);
    int min = arr[0];
    for (int j = 1; j < i; j++)
    {
        if(arr[j] < min){
            min = arr[j];
        }
    }
    int main_arr[i-1];
    int k = 0;
    for (int j = 0; j < i; j++)
    {
        if(arr[j] != min){
            main_arr[k] = arr[j];
            k++;
        }
    }
    int res = 1;
    for (int j = 0; j < i - 1; j++)
    {
        res = lcm(res, main_arr[j]);

    }
    if (isPrime(res+min))
    {
        printf("%d", res+min);
    }
    else{
        printf("None");
    }
    

    return 0;
}

File: CodeVita/prime.py
def gcd(a, b):
    if(b == 0):
        return a
    return gcd(b, a%b)

def isprime(a):
    if(a == 1):
        return True
    elif(a == 2 or a == 3):
        return True
    elif(a%2 == 0 or a%3 == 0):
        return False
    i = 5
    while(i*i<a):
        if(a%i == 0 or a%(i+2) == 0):
            return False
        i += 6
    return True

def lcm(a, b):
    return (a*b)/gcd(a,b)

a = input().split()
a = [int(i) for i in a]

min = min(a)
res = 1
a.remove(min)
for i in a:
    res = lcm(res, i)
res = int(res)
i = 1
while(i*res < 10*10):
    if(isprime(min+res*i)):
        print(res*i+min)
        
        break
    i+=1
else:
    print("None")